keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1237 @@
+    of aggregating container level spec values.
+
+#### Eviction Manager
+
+For memory.available signal, the eviction manager checks the pod's memory usage and compares it with memory requests at pod level to determine the pod eviction order. It aggregates the 
+container-level memory requests to calculate the effective pod-level memory request.When pod-level memory requests are specified, the Eviction Manager should use the pod-level request 
+directly instead of aggregating container-level requests. This logic should be modified to check if pod-level requests are set. If so, use the pod-level request; otherwise, fall back 
+to aggregating container requests.
+
+
+#### PodOverhead
+
+Since the PodOverhead feature works at pod-level cgroup, pod-level resource specification and PodOverhead will work together seamlessly. When the kube-scheduler is deciding 
+which node should run a new Pod, the scheduler considers that Pod's overhead as well as the sum of container requests for that Pod. Scheduler should add pod-level request,
+podoverhead to find a suitable node, when pod level request is specfied.
+
+Likewise the kubelet will need to add PodOverhead to the limits specified in the pod-level spec (instead of sum of container level requests) when setting the upper limit for
+the pod-level cgroup (i.e. cpu.max and memory.max).
+
+
+#### [Scoped for Beta] HugeTLB cgroup
+
+Note: This section includes only high level overview; Design details will be added in Beta stage.
+
+To support pod-level resource specifications for hugepages, Kubernetes will need to adjust how it handles hugetlb cgroups. Unlike memory, where an unset limit 
+means unlimited, an unset hugetlb limit is the same as setting it to 0.
+
+With the proposed changes, hugepages-2Mi and hugepages-1Gi will be added to the pod-level resources section, alongside CPU and memory. The hugetlb cgroup for the
+pod will then directly reflect the pod-level hugepage limits, rather than using an aggregated value from container limits. When scheduling, the scheduler will 
+consider hugepage requests at the pod level to find nodes with enough available resources.
+
+
+#### [Scoped for Beta] Topology Manager
+
+Note: This section includes only high level overview; Design details will be added in Beta stage.
+
+
+* (Tentative) Only pod level scope for topology alignment will be supported if pod level requests and limits are specified without container-level requests and limits.
+* The pod level scope for topology aligntment will consider pod level requests and limits instead of container level aggregates.
+* The hint providers will consider pod level requests and limits instead of container level aggregates.
+
+
+#### [Scoped for Beta] Memory Manager
+
+TBD. Do not review for the alpha stage.
+
+
+#### [Scoped for Beta] CPU Manager
+
+TBD. Do not review for the alpha stage.
+
+
+#### [Scoped for Beta] In Place Pod Resize
```

Comment:
To do in-place resize, we almost certainly need status fields added, and new fields always start as alpha.  So our choices are: do it in this alpha (or at least add the fields), do in in 1.32 but stay alpha, or wait for this to go beta and then add another alpha feature-gate to cover in-place updates of pod-level resources.

@tallclair

Reply:
@tallclair not sure if we ended up discussing this. If we haven't, we should finalize this ...